src/tests/unit_x509.cpp

```diff
@@ -1592,8 +1593,885 @@ Test::Result test_x509_tn_auth_list_extension_decode() {
    return result;
 }
 
+Test::Result test_x509_ip_addr_blocks_extension_decode() {
+   Test::Result result("X509 IP Address Block decode");
+   result.start_timer();
+   const std::string filename("IPAddrBlocksAll.pem");
+
+   Botan::X509_Certificate cert(Test::data_file("x509/x509test/" + filename));
+
+   using Botan::Cert_Extension::IPAddressBlocks;
+
+   auto ip_addr_blocks = cert.v3_extensions().get_extension_object_as<IPAddressBlocks>();
+
+   const auto& addr_blocks = ip_addr_blocks->addr_blocks();
+   result.confirm("cert has IPAddrBlocks extension", ip_addr_blocks != nullptr, true);
+   result.confirm("cert has exactly one IpAddrBlock", addr_blocks.size() == 2, true);
+
+   const auto& ipv4block =
+      std::get<IPAddressBlocks::IPAddressChoice<IPAddressBlocks::Version::IPv4>>(addr_blocks[0].addr_choice());
+   const auto& ipv6block =
+      std::get<IPAddressBlocks::IPAddressChoice<IPAddressBlocks::Version::IPv6>>(addr_blocks[1].addr_choice());
+
+   auto& v4_blocks = ipv4block.ranges().value();
+
+   result.confirm("ipv4 block 0 min", v4_blocks[0].min().value() == std::array<uint8_t, 4>{192, 168, 0, 0}, true);
```

Comment:
For comparing octet strings in tests I generally prefer

`result.test_eq("foo equals bar", hex_encode(binary_data), "C0A80000")`

since then if it doesn't match you get a message like

"ABCDEF00 != C0A80000"

rather than, effectively

"it didn't match"

Reply:
Oh I forgot there is an existing helper for this in the tests

`result.test_eq("foo equals bar", binary_data, "00010203");`